src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

```diff
@@ -200,23 +206,57 @@ public void getHtmlOutputForSubmission(@PathVariable Long submissionId, @Request
 		response.setHeader(HttpHeaders.CONTENT_DISPOSITION, "inline");
 		response.setHeader(HttpHeaders.CONTENT_TYPE, "text/html");
 
-		try (InputStream inputStream = new FileInputStream(outputFile.getFile()
-				.toString()); OutputStream outputStream = response.getOutputStream()) {
-			// Copy the file contents to the response outputstream
-			IOUtils.copy(inputStream, outputStream);
-		} catch (IOException e) {
-			logger.debug("Html output not found.");
-			String htmlOutputNotFound = messageSource.getMessage("analysis.html.file.not.found",
-					new Object[] { filename }, locale);
-			OutputStream outputStream = response.getOutputStream();
-			/*
-			Write the htmlNotFound message to the outputstream. We do this
-			so that the page doesn't error and will instead display the
-			message.
-			 */
-			outputStream.write(htmlOutputNotFound.getBytes(StandardCharsets.UTF_8));
-			outputStream.flush();
-			outputStream.close();
+		if (FileUtilities.isZippedFile(outputFile.getFile())) {
```

Comment:
I'm wondering if you could add in a check for `.zip` file extension in addition to using `isZippedFile()` here? Or maybe check for a `.html` extension (if we chose to not rename `.html` to `.html.zip`)?

The reason being is that some files (like Excel `xlsx`) are actually zip files with a renamed extension and we don't want to open them up and look for HTML files.

Reply:
This function is now just being call inside of `FileUtilities.getFileExt` when it determines that ext is `html` from regex. Then if it is zipped html `FileUtilities.getFileExt` will return `html-zip`.

Reply:
Awesome. Sounds great.